### ISSUE TITLE
Add min-p sampling to flare-core and WASM API

### ISSUE THREAD

**Issue by sauravpanda** [sauravpanda/flarellm#219]
Min-P sampling is a modern alternative to Top-P that samples all tokens whose probability exceeds `min_p * p_max` (where `p_max` is the max token probability). It avoids the "repetition trap" that nucleus sampling can fall into and generally produces higher quality output.

## Acceptance Criteria
- Add `min_p: f32` field to `SamplingParams` in `flare-core/src/sampling.rs` (default 0.0 = disabled)
- Add `sample_min_p(logits, min_p, rng) -> u32` function in `flare-core/src/sampling.rs`
- Update `Generator::step()` to apply min_p when `min_p > 0` and `top_p == 1.0` and `top_k == 0`
- Expose `min_p: f32` parameter in `begin_stream_with_params` and `generate_with_params` on `FlareEngine` (7th parameter)
- Update `generate_stream_with_params` (issue #218) to also accept min_p

## Algorithm
```
probs = softmax(logits)
threshold = min_p * max(probs)
candidates = [(i, p) for (i, p) in enumerate(probs) if p >= threshold]
# sample from candidates proportionally
```

## Notes
- No unsafe code needed
- Purely additive change to SamplingParams (backward compatible via Default)
